### PR TITLE
Added threading logic, refactored RefreshAction and fixed checkstyle…

### PR DESCRIPTION
**Pull request by amit-shar**
…, findbugs issue

### CODE REVIEW

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java [hunk 1]
```diff
@@ -81,85 +88,153 @@ public RefreshAction() {
...
+            ie.printStackTrace();
+        } catch (ExecutionException ee) {
+            ee.printStackTrace();
         }
     }
-
-    public final void runAction(final AnActionEvent anActionEvent) throws IOException {
-        Project project = windowObjects.getProject();
-        final Editor projectEditor = FileEditorManager.getInstance(project).getSelectedTextEditor();
+    private void setWindowObjects(@NotNull final AnActionEvent anActionEvent) {
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java]: Which window object is this ? Can we have proper names please ?

File: plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java [hunk 2]
```diff
@@ -235,9 +317,26 @@ private void createEditor(final JPanel editorPanel, final String contents) {
         Editor tinyEditor =
                 EditorFactory.getInstance().
                         createEditor(tinyEditorDoc, project, fileType, false);
-
         editorPanel.add(tinyEditor.getComponent());
         editorPanel.revalidate();
         editorPanel.repaint();
     }
+
+    private class Worker implements Runnable {
```
Comment (ScrapCodes) [inline plugins/idea/betterdocsidea/src/main/java/com/imaginea/betterdocs/RefreshAction.java]: What worker ? What does it do ?

**Comment by ScrapCodes** [comment]
Can you update you, so that I can merge it ?

**Comment by amit-shar** [comment]
This has been taken care in #119 . So closing this PR

**Pull request closed without merging**